test/e2e/handler/nncp_cleanup_test.go

```diff
@@ -15,24 +15,24 @@ import (
 
 var _ = Describe("NNCP cleanup", func() {
 
-	BeforeEach(func() {
-		By("Create a policy")
-		setDesiredStateWithPolicy(bridge1, linuxBrUp(bridge1))
+	Context("when a policy is deleted", func() {
+		BeforeEach(func() {
+			By("Create a policy")
+			setDesiredStateWithPolicy(bridge1, linuxBrUp(bridge1))
 
-		By("Wait for policy to be ready")
-		waitForAvailablePolicy(bridge1)
-	})
+			By("Wait for policy to be ready")
+			waitForAvailablePolicy(bridge1)
 
-	AfterEach(func() {
-		updateDesiredState(linuxBrAbsent(bridge1))
-		waitForAvailableTestPolicy()
-		resetDesiredStateForNodes()
-	})
+			By("Delete the policy")
+			deletePolicy(bridge1)
```

Comment:
Why do you need to call deletePolicy in both BeforeEach and AfterEach?

Reply:
In case something fails AfterEach will remove it (or try to)

Reply:
In case something goes sideways at BeforeEach and deletePolicy is not reached AfterEach will clean it.